api/repositories/app_repository_test.go

```diff
@@ -183,214 +178,178 @@ var _ = Describe("AppRepository", func() {
 
 		BeforeEach(func() {
 			message = ListAppsMessage{}
 
-			var cfAppList workloadsv1alpha1.CFAppList
-			Expect(
-				k8sClient.List(context.Background(), &cfAppList),
-			).To(Succeed())
-
-			for _, app := range cfAppList.Items {
-				Expect(
-					k8sClient.Delete(context.Background(), &app),
-				).To(Succeed())
-			}
+			space2 := createSpaceWithCleanup(testCtx, org.Name, prefixedGUID("space2"))
+			space3 := createSpaceWithCleanup(testCtx, org.Name, prefixedGUID("space3"))
+			createRoleBinding(testCtx, userName, spaceDeveloperRole.Name, space.Name)
+			createRoleBinding(testCtx, userName, spaceDeveloperRole.Name, space2.Name)
 
-			nonCFNamespace = prefixedGUID("non-cf")
-			Expect(k8sClient.Create(
-				testCtx,
-				&corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: nonCFNamespace}},
-			)).To(Succeed())
+			cfApp2 = createApp(space2.Name)
+			createApp(space3.Name)
+		})
 
-			createRoleBinding(testCtx, userName, spaceDeveloperRole.Name, space1.Name)
-			createRoleBinding(testCtx, userName, spaceDeveloperRole.Name, space2.Name)
-			createRoleBinding(testCtx, userName, spaceDeveloperRole.Name, nonCFNamespace)
+		JustBeforeEach(func() {
+			var err error
+			appList, err = appRepo.ListApps(testCtx, authInfo, message)
+			Expect(err).NotTo(HaveOccurred())
 		})
 
-		Describe("when filters are NOT provided", func() {
-			When("no Apps exist", func() {
-				It("returns an empty list of apps", func() {
-					appList, err := appRepo.ListApps(testCtx, authInfo, message)
-					Expect(err).NotTo(HaveOccurred())
-					Expect(appList).To(BeEmpty())
-				})
+		It("returns all the AppRecord CRs where client has permission", func() {
+			Expect(appList).To(ConsistOf(
+				MatchFields(IgnoreExtras, Fields{"GUID": Equal(cfApp.Name)}),
+				MatchFields(IgnoreExtras, Fields{"GUID": Equal(cfApp2.Name)}),
+			))
+
+			sortedByName := sort.SliceIsSorted(appList, func(i, j int) bool {
+				return appList[i].Name < appList[j].Name
 			})
 
-			When("multiple Apps exist", func() {
-				BeforeEach(func() {
-					cfApp1 = createApp(space1.Name)
-					cfApp2 = createApp(space2.Name)
-					createApp(space3.Name)
-					createApp(nonCFNamespace)
-				})
+			Expect(sortedByName).To(BeTrue(), fmt.Sprintf("AppList was not sorted by Name : App1 : %s , App2: %s", appList[0].Name, appList[1].Name))
+		})
 
-				It("returns all the AppRecord CRs where client has permission", func() {
-					appList, err := appRepo.ListApps(testCtx, authInfo, message)
-					Expect(err).NotTo(HaveOccurred())
-					Expect(appList).To(ConsistOf(
-						MatchFields(IgnoreExtras, Fields{"GUID": Equal(cfApp1.Name)}),
-						MatchFields(IgnoreExtras, Fields{"GUID": Equal(cfApp2.Name)}),
-					))
+		When("there are apps in non-cf namespaces", func() {
+			var nonCFApp *workloadsv1alpha1.CFApp
 
-					sortedByName := sort.SliceIsSorted(appList, func(i, j int) bool {
-						return appList[i].Name < appList[j].Name
-					})
+			BeforeEach(func() {
+				nonCFNamespace := prefixedGUID("non-cf")
```

Comment:
The non-cf namespace needs to be cleaned up in an AfterEach

Reply:
Why? Env tests generally do not need clean up as their environment is thrown away anyway. How about getting rid of all cleanup?